frontend-react/src/components/ManagePublicKey/ManagePublicKey.tsx

```diff
@@ -82,50 +91,77 @@ function ManagePublicKeySwitchDisplay() {
         }
     };
 
+    const showPublicKeyConfigured =
+        sender.length > 0 &&
+        hasPublicKey &&
+        !uploadNewPublicKey &&
+        !fileSubmitted;
+    const showUploadMsg =
+        (sender.length > 0 && !fileSubmitted && !hasPublicKey) ||
+        (!uploadNewPublicKey && sender.length === 0);
+    const showUpload =
+        (sender.length > 0 && !fileSubmitted && !hasPublicKey) ||
+        uploadNewPublicKey;
+    const showUploadError = fileSubmitted && !isUploading && !isSuccess;
+
     return (
-        <>
-            {/*Waiting on backend to support this
+        <GridContainer className="manage-public-key padding-bottom-5 tablet:padding-top-6">
+            {!isUploading && (
+                <h1 className="margin-top-0 margin-bottom-3">
+                    Manage public key
+                </h1>
+            )}
+            {showUploadMsg && (
+                <>
+                    <p className="font-sans-md">
+                        Send your public key to begin the REST API
+                        authentication process.
+                    </p>
+                    <SiteAlert variant="info" showIcon={false}>
+                        <Icon.Lightbulb />
+                        <span className="padding-left-1">
+                            If you need more information on generating your
+                            public key, reference page 7 in the{" "}
+                            <USLink href="/resources/programmers-guide">
+                                API Programmer’s Guide.
+                            </USLink>
+                        </span>
+                    </SiteAlert>
+                </>
+            )}
             {sender.length === 0 && (
```

Comment:
Once the ManagePublicKeyChooseSender is made dumber to accept senders, we could check to see if there are more than 1 senders for the org and show this component. If there is only one, we could technically automatically set it as the upload sender.